src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferService.java

```diff
@@ -1,47 +1,210 @@
+              .requestId(UUID.randomUUID().toString())
+              .timestamp(Utils.getCurrentTimeStamp())
+              .error(errorResponse)
+              .resp(responseRequestId)
+              .build();
+    }
+    try {
+      log.info(hipConsentNotificationResponse.toString());
+      ResponseEntity<GatewayGenericResponse> response =
+          requestManager.fetchResponseFromGateway(dataOnNotifyPath, hipConsentNotificationResponse);
+      log.debug(dataOnNotifyPath + " : dataOnNotify: " + response.getStatusCode());
+      if (response.getStatusCode() == HttpStatus.ACCEPTED) {
+        requestLogService.dataTransferNotify(
+            hipConsentNotification,
+            RequestStatus.DATA_ON_NOTIFY_SUCCESS,
+            hipConsentNotificationResponse);
+      } else if (Objects.nonNull(response.getBody())
+          && Objects.nonNull(response.getBody().getErrorResponse())) {
+        requestLogService.dataTransferNotify(
+            hipConsentNotification,
+            RequestStatus.DATA_ON_NOTIFY_ERROR,
+            hipConsentNotificationResponse);
+      }
+    } catch (Exception ex) {
+      String error =
+          "Exception while Initiating dataTransfer onNotify: "
+              + ex.getMessage()
+              + " unwrapped exception: "
+              + Exceptions.unwrap(ex);
+      log.debug(error);
+    }
   }
 
   /**
-   *
    * POST /on-request as an acknowledgement for agreeing to make dataTransfer to ABDM gateway.
-   * @param HIPHealthInformationRequest HIU public keys and dataPush URL is provided
+   *
+   * @param hipHealthInformationRequest HIU public keys and dataPush URL is provided
    */
   @Override
-  public void requestOnReceived(HIPHealthInformationRequest HIPHealthInformationRequest) {
-    // TODO Implement onRequest workflow logic.
-    initiateBundleRequest(HIPHealthInformationRequest);
+  public void
+  requestOnReceived(HIPHealthInformationRequest hipHealthInformationRequest) {
+    HIPHealthInformationRequestAcknowledgement hipHealthInformationRequestAcknowledgement = null;
+    RespRequest responseRequestId =
+        RespRequest.builder().requestId(hipHealthInformationRequest.getRequestId()).build();
+    String consentId = hipHealthInformationRequest.getHiRequest().getConsent().getId();
+    HiRequest hiRequest =
+        HiRequest.builder()
+            .sessionStatus("ACKNOWLEDGED")
+            .transactionId(hipHealthInformationRequest.getTransactionId())
+            .build();
+    if (logsRepo.findByConsentId(consentId) != null) {
+      hipHealthInformationRequestAcknowledgement =
+          HIPHealthInformationRequestAcknowledgement.builder()
+              .requestId(UUID.randomUUID().toString())
+              .timestamp(Utils.getCurrentTimeStamp())
+              .hiRequest(hiRequest)
+              .resp(responseRequestId)
+              .build();
+    } else {
+      errorResponse.setMessage("HIP -> ConsentId not found in cache");
```

Comment:
Question: Should the error messages be passed in the callback API for Gateway i.e. /on-request rather than in the same API?